src/section/section_authority_provider.rs

```diff
@@ -61,6 +65,18 @@ impl SectionAuthorityProvider {
     pub(crate) fn position(&self, name: &XorName) -> Option<usize> {
         self.elders.keys().position(|other_name| other_name == name)
     }
+
+    pub(crate) fn set_pk_set(&mut self, pk_set: ReplicaPublicKeySet) {
+        self.pk_set = Some(pk_set);
+    }
+
+    pub(crate) fn elders_only(&self) -> Self {
```

Comment:
I think the name is misleading, I was about to ask "but the SAP is supposed to contain only elders!?", so maybe call it `clone_without_pk_set` ?